drivers/linstorvolumemanager.py

```diff
@@ -196,18 +200,57 @@ def _get_controller_uri():
         # Not found, maybe we are trying to create the SR...
         pass
 
-def get_controller_uri():
-    retries = 0
-    while True:
+
+def get_cached_controller_uri():
+    try:
+        with open(CONTROLLER_CACHE_PATH, "r") as f:
+            return f.read().strip()
+    except Exception as e:
+        util.SMlog('Unable to read controller uri cache file at {} : {}'.format(
+            CONTROLLER_CACHE_PATH,
+            e
+        ))
```

Comment:
```suggestion
        util.SMlog('Unable to read controller uri cache file `{}`: {}'.format(
            CONTROLLER_CACHE_PATH,
            e
        ))
```

Reply:
Also: I'm not sure that's a good idea to log if the file is missing. Maybe it's better to log in case of a real error like a disk failure.

Reply:
We could also log them in debug only
```suggestion
        util.SMlog('Unable to read controller uri cache file at {} : {}'.format(
            CONTROLLER_CACHE_PATH,
            e
        ), priority=util.LOG_DEBUG)
```

Reply:
fixed in [cbdf27c](https://github.com/xcp-ng/sm/pull/83/commits/cbdf27cdd10fc2c00fcfbe003607524b8fa9af7c). I also set the priority as debug as Damien suggested.

Reply:
Not sure that's a good idea to hide important info using debug logs. In fact if we can't read/open an existing file, it should be a real error. Same for the write case if we can't open/save.

Reply:
Ok, I will change it to add error logs in the event of a real/severe error

Reply:
I adjusted it by adding a more granular error handling in [bc6c822](https://github.com/xcp-ng/sm/pull/83/commits/bc6c822cbf854c4ec7b1d977f55af4296b5c446e)

The idea here is to log more common errors in debug meanwhile OS errors or un-handled errors are logged normally